### PR TITLE
Remove deprecated way of calling ``esmvalcore.cmor.table.read_cmor_tables``

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

Remove a deprecated way of calling [``esmvalcore.cmor.table.read_cmor_tables``](https://docs.esmvaltool.org/projects/ESMValCore/en/v2.9.0/api/esmvalcore.cmor.html#esmvalcore.cmor.table.read_cmor_tables). The function argument should now be a config-developer.yml file (and not a ``dict`` with the content of the file). 

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [ ] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [ ] ~[🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)~
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: tests/integration/cmor/test_read_cmor_tables.py [hunk 1]
```diff
@@ -1,6 +1,5 @@
 from pathlib import Path
 
-import pytest
```
Comment (valeriupredoi) [inline tests/integration/cmor/test_read_cmor_tables.py:3]: think you still need this one for `tmp_path`

**Review by valeriupredoi — CHANGES_REQUESTED** [review]
tiny wiggle, have a quick look pls bud :beer:

File: tests/integration/cmor/test_read_cmor_tables.py (hunk 1 above)
Comment (bouweandela) [reply tests/integration/cmor/test_read_cmor_tables.py:3]: It's not needed, fixtures are automatically injected.

Comment (bouweandela) [reply tests/integration/cmor/test_read_cmor_tables.py:3]: To expand on that: if it were needed, the tests would be failing ;-)

Comment (valeriupredoi) [reply tests/integration/cmor/test_read_cmor_tables.py:3]: yeah I was wondering why it hasn't so far :laughing:

**Review by valeriupredoi — APPROVED** [review]
you out-fixtured me @bouweandela :grin:

**Pull request merged**